### PR TITLE
Support for serializing ResourceList.results field from functions

### PR DESCRIPTION
**Pull request by pwittrock**

### CODE REVIEW

**Comment by pwittrock** [comment]
@frankfarzan

File: kyaml/kio/byteio_readwriter_test.go [hunk 1]
```diff
@@ -0,0 +1,263 @@
...
+`,
+		},
+
+		{
+			name: "function_config",
+			input: `
+apiVersion: config.kubernetes.io/v1alpha1
+kind: ResourceList
+items:
+- kind: Deployment
```
Comment (phanimarupaka) [inline kyaml/kio/byteio_readwriter_test.go:59]: Add a comment above this line and see if it's being parsed correctly.

File: kyaml/kio/byteio_readwriter_test.go [hunk 2]
```diff
@@ -0,0 +1,263 @@
...
+  spec:
+    replicas: 1
+- kind: Service
+  spec:
+    selectors:
+      foo: bar
+foo:
```
Comment (phanimarupaka) [inline kyaml/kio/byteio_readwriter_test.go:130]: Why is this being dropped ?

Comment (pwittrock) [reply kyaml/kio/byteio_readwriter_test.go:130]: because it isn't a valid field in ResourceList

File: kyaml/kio/byteio_readwriter_test.go (hunk 1 above)
Comment (pwittrock) [reply kyaml/kio/byteio_readwriter_test.go:59]: it isn't serialized correctly, lets follow up with why that is

File: kyaml/kio/filters/container.go [hunk 3]
```diff
@@ -314,6 +326,25 @@ func (c *ContainerFilter) Filter(nodes []*yaml.RNode) ([]*yaml.RNode, error) {
...
 }
 
+func (c *ContainerFilter) doResults(r *kio.ByteReader) error {
+	// Write the results to a file if configured to do so
+	if c.ResultsFile != "" && r.Results != nil {
+		results, err := r.Results.String()
+		if err != nil {
+			return err
+		}
+		err = ioutil.WriteFile(c.ResultsFile, []byte(results), 0600)
```
Comment (phanimarupaka) [inline kyaml/kio/filters/container.go:336]: Maybe create a dir if it doesn't exist.

**Pull request merged**